src/stats/probs/adaptive_integration.rs

```diff
@@ -112,7 +113,7 @@ where
     // METHOD additionally investigate small interval around the optimum
     for point in linspace(
         cmp::max(
-            middle.unwrap() - (max_resolution.into() * 3.0).into(),
+            middle.unwrap() - T::try_from(max_resolution.into() * 3.0).unwrap(),
```

Comment:
I would make it
```suggestion
            T::try_from(middle.unwrap() - max_resolution.into() * 3.0).unwrap(),
```
in order to correctly handle cases of overflow/underflow.

Reply:
For NotNan, yes.  But this is a generic function and `T` could be a type which didn't allow `Inf` either, for example.  All you are supposed to know here is that the conversion from `f64` to `T` is fallible.

Reply:
Likely not worth spending much time on this particular issue, but since I'm still learning Rust wanted to pick your brain with a little more discussion.

My understanding of this code is that both operands to the subtraction, `middle.unwrap()`  and `T::try_from(max_resolution.into() * 3.0)` are of type T.  I also believe that because the method has the constraint that `T: Sub<Output = T>`, the result of this subtraction should be of type `T`.

If so, then I don't see any benefit to this change, if `T` is a type that isn't supposed to allow `Inf`, but I can still construct an instance of it that has an `Inf` value through subtraction, it feels like `T` doesn't have the guarantees it thinks it does and using `try_from` is just a band-aid around that fundamental issue.